### PR TITLE
#9492: move matmul code to ttnn directory hierarchy

### PR DESCRIPTION
**Pull request by bbradelTT**
### Ticket
Link to Github Issue https://github.com/tenstorrent/tt-metal/issues/9492

### Problem description
- We're moving ops over to ttnn

### What's changed
- Moving over matmul
- Moved over bindings for matmul program configs

### Checklist
- [x] Post commit CI passes
- [x] Model regression CI testing passes (if applicable) N/A
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tests/tt_eager/integration_tests/test_bert.cpp [hunk 1]
```diff
@@ -35,10 +35,12 @@ Tensor encoder(Tensor&& hidden_states, const Tensor& attention_mask, const Param
         .transpose_mcast = false,
         .fused_activation = std::nullopt,
     };
-    auto fused_qkv_matmul_output = tt::operations::primary::matmul(
+    auto fused_qkv_matmul_output = ttnn::operations::matmul::matmul(
         hidden_states,
         parameters.at(fmt::format("fused_qkv_weight_{}", encoder_index)),
         parameters.at(fmt::format("fused_qkv_bias_{}", encoder_index)),
+        /*transpose_a=*/false,
+        /*transpose_b=*/false,
```
Comment (ayerofieiev-tt) [inline tests/tt_eager/integration_tests/test_bert.cpp:43]: can we use enum flags instead?

So one can do 
`matmul(hiddent_states, params1, params2, Transpose::Both/None/A/B...)`
or
`matmul(hiddent_states, params1, params2, Transpose::A | Transpose::B)`

**Review by ayerofieiev-tt — APPROVED** [review]

File: tests/tt_eager/integration_tests/test_bert.cpp (hunk 1 above)
Comment (ayerofieiev-tt) [reply tests/tt_eager/integration_tests/test_bert.cpp:43]: (not a matter of this PR)

**Review by yugaoTT — APPROVED** [review]
lgtm

**Review by TT-BrianLiu — APPROVED** [review]
Beautiful

File: tests/tt_eager/integration_tests/test_bert.cpp (hunk 1 above)
Comment (bbradelTT) [reply tests/tt_eager/integration_tests/test_bert.cpp:43]: Definitely not a matter for this PR.
The parameters were added as part of https://github.com/tenstorrent/tt-metal/pull/9836

**Pull request merged**